### DISCUSSION TITLE
UTF-32 클래스 만들기

### DISCUSSION THREAD

**Discussion by ycs1m1yk** [euphony-io/euphony#105] [Ideas]
UTF-32 클래스는 UTF-16 클래스를 약간 응용하면 만들 수 있습니다.

https://github.com/euphony-io/euphony/blob/33c6f19b3bc07bd29a6b4edaccdf8fe77574c275/euphony/src/main/cpp/core/UTF16Charset.h#L8-L20

UTF16Charset 헤더입니다.

privite 영역에 encode, decode 메소드를 정의하는데 필요한 값들이 정의되어 있습니다.

UTF-32는 모든 문자를 4바이트로 표현합니다. 

그에 맞춰서 재정의하면 아래와 같습니다.

```cpp
private:
        const u_int8_t BIT_COUNT_IN_HEX_NUM = 4;
        const u_int8_t BYTE_COUNT = 4;
        const u_int8_t HEX_NUM_COUNT = BYTE_COUNT * 8 / BIT_COUNT_IN_HEX_NUM;
        const char32_t BIT_MASK = 0x0000000F;
```

**Comment by ycs1m1yk** [comment]
# *작성중

# encode 메소드 구현하기

https://github.com/euphony-io/euphony/blob/33c6f19b3bc07bd29a6b4edaccdf8fe77574c275/euphony/src/main/cpp/core/source/UTF16Charset.cpp#L7-L21

UTF16Charset의 encode 메소드 구현입니다.

### 키워드
1.  harfByteBuffer? -> harfByte 보다는 hex / hexadecimal / base16 중에서
2.  bit연산 활용

**Comment by ycs1m1yk** [comment]
# *작성중

# decode 메소드 구현하기

https://github.com/euphony-io/euphony/blob/33c6f19b3bc07bd29a6b4edaccdf8fe77574c275/euphony/src/main/cpp/core/source/UTF16Charset.cpp#L26-L43

UTF16Charset의 decode 메소드 구현입니다.

### 키워드
1.  codecvt
2.  convert? converter? 명명
3.  가독성

**Comment by ycs1m1yk** [comment]
# 테스트케이스

유포니 여러분의 집단지성  #95 을 활용할 것입니다.

덕분에 다양하게 테스트해볼 수 있네요! 👍 👍

**Comment by kuro11pow2** [comment]
고생하셨습니다! 👍

**Comment by aiclaudev** [comment]
수고하셨습니다! 😃
